mitodl/open-discussions
# List Items API

#### Pre-Flight checklist

#### What are the relevant tickets?
- Partially addresses #2464

#### What's this PR do?
- Adds a nested UserListItems API

#### How should this be manually tested?
- All the following should work the same as before from the UI:
  - Adding/removing items from lists
  - Creating/updating/deleting lists

New functionality:
- Go to `http://od.odl.local:8063/api/v0/userlists/<user_list_id>/items/` for one of your own lists or another user's public list, the response should include the items and their details in that list.
- Try the above for another user's private list, you should get a 403
- Try POSTing a new item, it should be created:
  ```
  { "content_type": "course", "object_id": <course_id>}
  ```
- Go to `http://od.odl.local:8063/api/v0/userlists/<user_list_id>/items/<item_id>/` for one of the items in your own list or another user's public list, the response should include the item details.
- Try the above for another user's private list, you should get a 403
- Try PATCHing the item with a new position, the position should be updated.
  ```
  { "position": 8675309}
  ```
- Try DELETEing the item, it should be deleted.
